src/baldr/graphreader.cc

```diff
@@ -295,12 +298,14 @@ TileCache* TileCacheFactory::createTileCache(const boost::property_tree::ptree&
       }
     }
     return new SynchronizedTileCache(*globalTileCache_, globalCacheMutex_);
-  }
+  } else {
```

Comment:
why add the `else`? to me its more confusing. the above logic cant fall through so it doesnt actually add anything but more braces and indents, right? i guess compilers are smart these days, but it also makes it so that there is no return statement in the outer most scope.

Reply:
I found it useful with the symmetry, one branch for the synchronization and one for the no synchronization case. I can strip out the second branch.

Reply:
I liked the symmetry of the synchronized and not-synchronized case. I believe the clang warnings would throw an error if it detected some branching path that led to no return value. I can remove the added `else` branch though.